src/main/java/com/es/phoneshop/model/product/recentlyview/DefaultRecentlyViewService.java

```diff
@@ -0,0 +1,58 @@
+package com.es.phoneshop.model.product.recentlyview;
+
+import com.es.phoneshop.model.product.Product;
+
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpSession;
+import java.util.List;
+import java.util.concurrent.locks.ReadWriteLock;
+import java.util.concurrent.locks.ReentrantReadWriteLock;
+
+public class DefaultRecentlyViewService implements RecentlyViewService {
+
+    public static final String RECENTLY_VIEW_SECTION_ATTRIBUTE = DefaultRecentlyViewService.class.getName() + ".recentlyViewSection";
+    private static volatile DefaultRecentlyViewService instance;
+    private final ReadWriteLock lock = new ReentrantReadWriteLock();
+
+    private DefaultRecentlyViewService() {
+    }
+
+    public static DefaultRecentlyViewService getInstance() {
+        DefaultRecentlyViewService result = instance;
+        if (result != null) {
+            return result;
+        }
+        synchronized (DefaultRecentlyViewService.class) {
+            if (instance == null) {
+                instance = new DefaultRecentlyViewService();
+            }
+            return instance;
+        }
+    }
+
+    @Override
+    public synchronized void add(RecentlyViewSection recentlyViewSection, HttpServletRequest request, Product product) {
+        int sizeOfRecentlyViewBar = 3;
+        recentlyViewSection.getRecentlyView().remove(product);
+        List<Product> recentlyView = recentlyViewSection.getRecentlyView();
```

Comment:
Initialize the variable with recentlyViewSection.getRecentlyView() value and then use it instead of calling recentlyViewSection.getRecentlyView() multiple times.